ground/src/main/java/com/google/android/ground/model/locationofinterest/LocationOfInterest.kt

```diff
@@ -22,37 +22,39 @@ import com.google.android.ground.model.mutation.LocationOfInterestMutation
 import com.google.android.ground.model.mutation.LocationOfInterestMutation.Companion.builder
 import com.google.android.ground.model.mutation.Mutation
 import com.google.android.ground.model.mutation.Mutation.SyncStatus
+import com.google.android.ground.util.toImmutableList
+import com.google.common.collect.ImmutableList
+import org.locationtech.jts.geom.Geometry
+import org.locationtech.jts.geom.Polygon
 import java.util.*
 import javax.annotation.OverridingMethodsMustInvokeSuper
 
-/** Base class for user-defined locations of interest shown on the map. */
-sealed class LocationOfInterest {
-    // TODO: Once all callers are converted to Kotlin, we don't need these properties.
-    val isPoint: Boolean
-        get() = this is PointOfInterest
-    val isPolygon: Boolean
-        get() = this is AreaOfInterest
-
+/** User-defined locations of interest (LOI) shown on the map. */
+data class LocationOfInterest(
     /** A system-defined ID for this LOI. */
-    abstract val id: String
-
+    val id: String,
     /** The survey associated with this LOI. */
-    abstract val survey: Survey
-
+    val survey: Survey,
     /** The job associated with this LOI. */
-    abstract val job: Job
-
+    val job: Job,
     /** A user-specified ID for this location of interest. */
-    abstract val customId: String?
-
+    val customId: String? = null,
     /** A human readable caption for this location of interest. */
-    abstract val caption: String?
-
+    val caption: String? = null,
     /** User and time audit info pertaining to the creation of this LOI. */
-    abstract val created: AuditInfo
-
+    val created: AuditInfo,
     /** User and time audit info pertaining to the last modification of this LOI. */
-    abstract val lastModified: AuditInfo
+    val lastModified: AuditInfo,
+    /** Geometry associated with this LOI. */
+    val geometry: Geometry,
```

Comment:
Looking at this now, wondering if it would be more type-safe to keep the AOI and POI classes, having AOI contain `Polygon` and POI contain `Point`?

Reply:
It is more type safe, but we'd also be unable to construct LOIs, which is fine if we're ok with keeping that class abstract.

Reply:
We spoke. A single concrete LOI class with `Geometry` will work; we can specialize into subclasses iff we need to down the road.